File: frontend/types/family-stats.ts
export type FamilyStatsRecipe = {
  recipe_id: number;
  title: string;
  category: string;
  image_url: string | null;
  count: number;
};

export type FamilyStatsOrderer = {
  user_id: number;
  username: string;
  count: number;
};

export type FamilyMonthlyStats = {
  month: string;
  completed_meals: number;
  dishes_made: number;
  total_orders: number;
  top_recipes: FamilyStatsRecipe[];
  top_orderers: FamilyStatsOrderer[];
};

export type FamilyMonthlyStatsResponse = {
  stats: FamilyMonthlyStats;
};

File: backend/app/schemas/family_stats.py
from pydantic import BaseModel, Field


class FamilyStatsRecipe(BaseModel):
    recipe_id: int
    title: str
    category: str
    image_url: str | None
    count: int


class FamilyStatsOrderer(BaseModel):
    user_id: int
    username: str
    count: int


class FamilyMonthlyStats(BaseModel):
    month: str
    completed_meals: int = 0
    dishes_made: int = 0
    total_orders: int = 0
    top_recipes: list[FamilyStatsRecipe] = Field(default_factory=list)
    top_orderers: list[FamilyStatsOrderer] = Field(default_factory=list)


class FamilyMonthlyStatsResponse(BaseModel):
    stats: FamilyMonthlyStats
